### ISSUE TITLE
tensorflow process hangs with use of cudnn_rnn

### ISSUE THREAD

**Issue by harryxu-yscz** [tensorflow/tensorflow#17968]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: v1.4.0-rc1-11-g130a514 1.4.0
- **Python version**: 3.6.2
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: 8.0.61/6.0.21
- **GPU model and memory**: 1080 8GB
- **Exact command to reproduce**:

### Describe the problem
I ran a RNN model with tensorflow.contrib.cudnn_rnn.CudnnLSTM. After some epochs, the process hangs and never respond, even with `ctrl + c`. I also observed the volatile gpu-util fixes at 100% after the hang. This was not a problem when I ran the RNN with tensorflow.contrib.rnn.LSTMCell

### Source code / logs
call stacks of all Threads is here: 
https://pastebin.com/sNxC6fWC

**Comment by asimshankar** [comment]
@protoget : Could you take a look?

@harryxu-yscz : Is there a short program you can share that reproduces the problem?

**Comment by protoget** [comment]
@asimshankar Sure. It would be helpful to have a  minimum reproducible code snippet. Cudnn LSTM uses more GPU memory than non-platform-dependent TF LSTM.

**Comment by harryxu-yscz** [comment]
Unfortunate I am not able to provide a snippet that reproduces the problem. I hope the call stacks are helpful. 

Strangely, my friend ran exactly the same code on his machine and had no problem with it. Am I running into the hardware issue here https://github.com/tensorflow/tensorflow/issues/1947?

The LSTM is written as:
```
        import tensorflow.contrib as tc
        # bi-lstm
        lstm_fw = tc.cudnn_rnn.CudnnLSTM(num_layers=layer_num,
                                         num_units=hidden_size,
                                         input_size=input_size,
                                         # dropout = 1 - keep_prob
                                         dropout=0.)
        lstm_bw = tc.cudnn_rnn.CudnnLSTM(num_layers=layer_num,
                                         num_units=hidden_size,
                                         input_size=input_size,
                                         # dropout = 1 - keep_prob
                                         dropout=0.)
        param_fw = tf.get_variable("lstm_fw_params",
                                   initializer=tf.random_uniform([lstm_fw.params_size()],
                                                                 -0.1, 0.1),
                                   validate_shape=False)
        param_bw = tf.get_variable("lstm_bw_params",
                                   initializer=tf.random_uniform([lstm_bw.params_size()],
                                                                 -0.1, 0.1),
                                   validate_shape=False)
        c_fw = tf.zeros([layer_num, batch_size, hidden_size],
                        tf.float32)
        c_bw = tf.zeros([layer_num, batch_size, hidden_size],
                        tf.float32)
        h_fw = tf.zeros([layer_num, batch_size, hidden_size],
                        tf.float32)
        h_bw = tf.zeros([layer_num, batch_size, hidden_size],
                        tf.float32)
        with tf.variable_scope("fw"):
            out_fw, h_fw, c_fw = lstm_fw(inputs, h_fw, c_fw, param_fw)
        with tf.variable_scope("bw"):
            inputs_bw = tf.reverse_sequence(
                inputs, seq_lengths=length, seq_dim=0, batch_dim=1)
            out_bw, h_bw, c_bw = lstm_bw(inputs_bw, h_bw, c_bw, param_bw)
            out_bw = tf.reverse_sequence(
                out_bw, seq_lengths=length, seq_dim=0, batch_dim=1)
        output = tf.concat([out_fw, out_bw], axis=2)
        output = tf.transpose(output, [1, 0, 2])
        output_h = tf.concat([h_fw, h_bw], axis=0)
        output_c = tf.concat([c_fw, c_bw], axis=0)
        return output, (tc.rnn.LSTMStateTuple(h=output_h, c=output_c),)
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 105 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 120 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 120 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 135 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 150 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 165 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 180 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 195 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 210 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 225 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 240 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by simtony** [comment]
The same issue. It only happens on some runs. The process staled and the gpu utilization get 100%.

**Comment by milleramit** [comment]
Also happening with tf1.12.

Happens randomly, with low prob.
AWS P3 instances [happened on at least two different instances]. 
GPU at 100% "utilization", but temp/fan are low, and training is stuck.

**Comment by dodiak** [comment]
> Also happening with tf1.12.
> 
> Happens randomly, with low prob.
> AWS P3 instances [happened on at least two different instances].
> GPU at 100% "utilization", but temp/fan are low, and training is stuck.

The same problem here. And the job can not be killed. Any solution or explanation?
But I found out that the same is with Pytorch, so it is not Tensorflow related ...

**Comment by kumariko** [comment]
@harryxu-yscz We are checking to see if you still need help on this issue, as you are using an older version of tensorflow(1.x) which is officially considered as end of life. We recommend that you upgrade to 2.6 which is latest stable version of TF and let us know if the issue still persists in newer versions. Thanks!
